src/loaders/genome_collection/parse_computed_genome_attribs.py

```diff
@@ -133,16 +186,345 @@ def _read_tool_result(result_dir, batch_dir, kbase_collection, load_ver, tool_fi
+                                                trait_idx,
+                                                cell_count,
+                                                trait_type,
+                                                min_value,
+                                                max_value,
+                                                trait_val=trait_val)
+            visited_traits.add(trait_idx)
+            cell_count += 1
+
+        # fill in missing trait values with 0s for all cells
+        missing_trait_idxs = trait_idxs - visited_traits
+        for missing_trait_idx in missing_trait_idxs:
+            trait = _find_trait_by_id(missing_trait_idx, traits_meta)
+            trait_type = trait.get('trait_type')
+            min_value, max_value = _append_cell(heatmap_row,
+                                                missing_trait_idx,
+                                                cell_count,
+                                                trait_type,
+                                                min_value,
+                                                max_value,
+                                                trait_val=_SYS_DEFAULT_TRAIT_VALUE)
+            cell_count += 1
+
+        heatmap_rows.append(heatmap_row)
+
+    return heatmap_rows, min_value, max_value
+
+
+def _create_heatmap_objs(
+        traits_meta: dict[str, dict[str, str]],
+        traits_val: dict[str, list[dict[str, int | float | bool]]]) -> (HeatMapMeta, list[HeatMapRow]):
+    # Create the HeatMapMeta and list of HeatMapRow from parsed trait metadata and values
+
+    heatmap_rows, min_value, max_value = _parse_heatmap_rows(traits_meta, traits_val)
+    categories = _parse_categories(traits_meta)
+    heatmap_meta = HeatMapMeta(categories=categories, min_value=min_value, max_value=max_value)
+
+    return heatmap_meta, heatmap_rows
+
+
+def microtrait(root_dir, kbase_collection, load_ver):
+    """
+    Parse and formate result files as heatmap data generated by the MicroTrait tool.
+    """
+
+    result_dir = _locate_dir(root_dir, kbase_collection, load_ver, tool='microtrait')
+    batch_dirs = _get_batch_dirs(result_dir)
+
+    traits_meta, traits_val = dict(), dict()
+    for batch_dir in batch_dirs:
+        data_ids = [item for item in os.listdir(os.path.join(result_dir, batch_dir)) if
+                    os.path.isdir(os.path.join(result_dir, batch_dir, item))]
+        for data_id in data_ids:
+            data_dir = os.path.join(result_dir, batch_dir, data_id)
+            trait_count_file = os.path.join(data_dir, loader_common_names.TRAIT_COUNTS_FILE)
+            selected_cols = [_MICROTRAIT_TRAIT_NAME,
+                             _MICROTRAIT_TRAIT_DISPLAYNAME_SHORT,
+                             _MICROTRAIT_TRAIT_DISPLAYNAME_LONG,
+                             _MICROTRAIT_TRAIT_VALUE,
+                             _MICROTRAIT_TRAIT_TYPE,
+                             _MICROTRAIT_TRAIT_ORDER,
+                             ]
```

Comment:
👍

Reply:
So this change looks good, but the zip stuff I was talking about assumed you used the list of names outside of the dict construction. Since you don't and dicts insertion order is preserved I think you can revert the dict construction to how it was before

Reply:
👍

Reply:
:+1: